rswag-specs/lib/rswag/specs/swagger_formatter.rb

```diff
@@ -33,19 +33,26 @@ def example_group_finished(notification)
         return if metadata[:document] == false
         return unless metadata.key?(:response)
 
-        swagger_doc = @config.get_swagger_doc(metadata[:swagger_doc])
-
-        unless doc_version(swagger_doc).start_with?('2')
-          # This is called multiple times per file!
-          # metadata[:operation] is also re-used between examples within file
-          # therefore be careful NOT to modify its content here.
-          upgrade_request_type!(metadata)
-          upgrade_servers!(swagger_doc)
-          upgrade_oauth!(swagger_doc)
-          upgrade_response_produces!(swagger_doc, metadata)
-        end
+        swagger_docs = (metadata[:swagger_docs] || [metadata[:swagger_doc]])
+                       .map { |doc| @config.get_swagger_doc(doc) }
+
+        swagger_docs.each { |swagger_doc|
+          response = Marshal.load(Marshal.dump(metadata[:response]))
```

Comment:
It's not clear to me what this serialise/deserialise step is for? Also, it looks a _little_ bit boiler-plate-y having the same calls for each of the three items. Would something like `response, operation, path_item = %i[response operation path_item].map ...` be clearer?

As an aside, why are you pulling out these three things into variables anyway, as it seems you are only passing them into the method below? In other words, would it make sense to keep just the `metadata` as the argument and extract the things you need, maybe using a private utility method, where you need them?

Reply:
So I briefly explained in the comment as well, but essentially there are a lot of `destructive` operations applied on those 3 sub-hashes of the `metadata` and we want them to have their own hash references to protect against per-document related changes for each of them.

So for example if a method lower down will change the `response` or `path_item` or `operation` for `doc_1`, we don't want that to spill into `doc_2` just because it happened to change a reference to an exact same hash that was accidentally put in the both of them.